pyigm/cgm/cgmsurvey.py

```diff
@@ -149,7 +156,133 @@ def nsys(self):
+            return len(self._dict)
+        else:
+            return len(self.cgm_abs)
+
+    def add_ion_to_data(self, inp_ion):
+        """ Add columns for ion info (column density)
+        to the Table
+
+        Parameters
+        ----------
+        inp_ion
+
+        Returns
+        -------
+
+        """
+        from linetools.analysis import absline as ltaa
+        ion = inp_ion.replace(' ','')
+        # Check for previous
+        if 'flag_N_{:s}'.format(ion) in self._data.keys():
+            print("Ion data is already in the _data table")
+            return
+        # Loop on the systems
+        flagNs, Ns, sigNs = [], [], []
+        for key in self._dict.keys():
+            igm_comp = self._dict[key]['igm_sys']['components']
+            comps = [] # Allow for more than one
+            for comp in igm_comp.keys():
+                sion = comp.split('_')[0]
+                if sion == ion:
+                    if 'attrib' in igm_comp[comp].keys():
+                        attrib = igm_comp[comp]['attrib']
+                        attrib['sig_logN'] = np.array(attrib['sig_logN'])
+                        comps.append(attrib.copy())
+                    else: # Deprecated
+                        comps.append(dict(logN=igm_comp[comp]['logN'],
+                                          flag_N=igm_comp[comp]['flag_N'],
+                                          sig_logN=np.array([igm_comp[comp]['sig_logN']]*2)))
+            # Now sum em up
+            if len(comps) == 0:
+                flagNs.append(0)
+                Ns.append(0.)
+                sigNs.append(np.array([0.]*2))
+                continue
+            obj = dict(flag_N=comps[0]['flag_N'], logN=comps[0]['logN'], sig_logN=comps[0]['sig_logN'])
+            for comp in comps[1:]:
+                if comp['flag_N'] != 0:
+                    obj['flag_N'], obj['logN'], obj['sig_logN'] = ltaa.sum_logN(obj, comp)
+            # Save
+            flagNs.append(obj['flag_N'])
+            Ns.append(obj['logN'])
+            sigNs.append(obj['sig_logN'])
+        # Add to Table
+        self._data.add_column(Column(flagNs, name='flag_N_{:s}'.format(ion)))
+        self._data.add_column(Column(Ns, name='logN_{:s}'.format(ion)))
+        self._data.add_column(Column(sigNs, name='sig_logN_{:s}'.format(ion)))
+
+    def build_sys_from_dict(self, sys_name, append=False, llist=None, **kwargs):
+        tdict = self._dict[sys_name]
+        cgmsys = CGMAbsSys.from_dict(tdict, chk_vel=False, chk_sep=False, chk_data=False,
+                                 use_coord=True, use_angrho=True,
+                                     linelist=llist, **kwargs)
+        if append:
+            self.cgm_abs.append(cgmsys)
+        # Return
+        return cgmsys
+
+
+    def build_systems_from_dict(self, **kwargs):
+        for key in self._dict.keys():
+            cgmsys = self.build_sys_from_dict(key, **kwargs)
+            self.cgm_abs.append(cgmsys)
```

Comment:
I think this could be sped up considerably if we imported linelist and used that each time.   Otherwise, building the systems via this command takes quite a while...

Reply:
I see below that there's a linelist keyword for load_tarball() but can we just declare one before the loop is none is provided?

Reply:
it should be coming in from **kwargs (e.g. see from_json at the top of the Class).

I could add a warning checking to see if kwargs includes llist.
Am doing so.